### PR TITLE
Add rancher dashboard version to "About" page

### LINKED ISSUE #4217: Have a way to see the exact Dashboard UI version / commit hash in Rancher
**Detailed Description**
Provide a way to see the exact Dashboard UI version / commit hash - either on UI or somewhere on payload where it can be viewed in browser's Dev Tools.

**Context**

This is helpful when validating issues or using Rancher in development to know exactly what UI commit is used to be sure that expected changes are included or not when it's may be not immediately clear when looking at the UI (case in point - the discussion here https://github.com/rancher/dashboard/issues/3995#issuecomment-922369495).

**UPDATE: As per indication from UX/UI team, the link on the bottom-right of the side menu featuring the backend version is to be replaced by link to the same about page but with the text "About".**

To test:
- test in a `-head` build that the dashboard version shown on the about page is of the format `"${COMMIT_BRANCH} ${COMMIT}"`
- test tagged (i.e. release candidate) build build that the dashboard version shown on the about page is the git version tag

### PR DESCRIPTION
**Pull request by aalves08**
Fixes #4217 

- Add rancher dashboard version to "About" page
- Replace copy of "backend version" link to just "About"
- Remove unused code from `TopLevelMenu` component

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -24,7 +24,7 @@
     "test:ci": "jest --collectCoverage",
     "nuxt": "./node_modules/.bin/nuxt",
     "dev": "./node_modules/.bin/nuxt dev",
-    "mem-dev": "node --max-old-space-size=8192 ./node_modules/.bin/nuxt dev",
+    "mem-dev": "source ./scripts/version && node --max-old-space-size=8192 ./node_modules/.bin/nuxt dev",
```
Comment (richard-cox) [inline package.json:27]: This needs to be applied to other dev like targets

File: scripts/version [hunk 2]
```diff
@@ -12,8 +12,10 @@ GIT_TAG=${DRONE_TAG:-$(git tag -l --contains HEAD | head -n 1)}
 LAST_TAG=${GIT_TAG:-'v0.0.0'}
 
 if [[ -z "$DIRTY" && -n "$GIT_TAG" ]]; then
+    export DASHBOARD_VERSION=$GIT_TAG
```
Comment (richard-cox) [inline scripts/version:15]: Thought about this for a while. I think we should rename or splash comments everywhere. It needs to be clear that `VERSION` is a more computer readable interpretation and `DASHBOARD_VERSION` is a more human readable one that can be used to display in the dashboard

File: shell/nuxt.config.js [hunk 3]
```diff
@@ -27,6 +27,8 @@ const version = process.env.VERSION ||
   process.env.DRONE_VERSION ||
   require('./package.json').version;
 
+const dashboardVersion = process.env.DASHBOARD_VERSION;
```
Comment (richard-cox) [inline shell/nuxt.config.js:30]: Same concept here, we need to ensure people understand the difference between version and dashboard version. I think though we could be brave and remove `version` and `commit` from here.

**Review by richard-cox — CHANGES_REQUESTED** [review]
Once this merges (and the drone build completes) we should validate that the dashboard version displays something sensible

File: scripts/version (hunk 2 above)
Comment (richard-cox) [reply scripts/version:16]: good comments, but also needs to apply to nuxt.config.js `version` and `dashboardVersion`

File: shell/nuxt.config.js (hunk 3 above)
Comment (richard-cox) [reply shell/nuxt.config.js:27]: Agree, they can both go

**Review by richard-cox — CHANGES_REQUESTED** [review]
One minor change requested.

You might need to add a note for QA to the issue, to test in a `-head` build and also a tagged (i.e. release candidate) build

**Comment by aalves08** [comment]
@richard-cox I've deleted the `version` on `shell/nuxt.config.js` and I've added the comment there. Let me know if I am missing something 🙏

**Pull request merged**